pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -55,10 +55,16 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	}
 	// Validate TaskSpec if it's present.
 	if ts.TaskSpec != nil {
-		errs = errs.Also(ts.TaskSpec.Validate(ctx).ViaField("taskSpec"))
+		skipVariableValidation := false
+		// this is for parameter propagation.
+		if config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields == "alpha" {
+			skipVariableValidation = true
+		}
```

Comment:
```suggestion
skipVariableValidation := (config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields == config.AlphaAPIFields)
```
instead of using "alpha" we can use config.AlphaAPIFields

Reply:
Good catch! I will fix that. Thanks